### PR TITLE
Modify `load_archived_murs` to properly handle MURs with multiple PDFs

### PR DESCRIPTION
**Pull request by rjayasekera**
- [X] Add test for nested PDFs

## Summary (required)
There are  9 Archived MURs that contain multiple pdf files.  We need to be able to load all associated PDF files when pass in a MUR number to `manage.py load_archived_murs` program with either -f (from MUR number) or -s (specific MUR number)

- One part of the resolution for https://github.com/fecgov/openFEC/issues/3234

_Include a summary of proposed changes._
Change the RegEx within `load_archived_murs` function in load_legal_docs.py to be able to pick up all parts of PDFs associated with a MUR number. Then in `process_murs` go thru a loop to load them one at a time to ES and S3.  

## How to test the changes locally
- Pull down the branch to local environment
- Point SQLA_CONN to DEV DB
- Set up S3 bucket variables
- if you don't already have it locally,  `./manage.py create_archived_murs_index`
- test loading MURS using -s and -f flags 
- ./manage.py load_archived_murs -s 2583
- run your local API and check http://localhost:5000/v1/legal/search/?mur_no= 2583&type=murs

## Impacted areas of the application
List general components of the application that this PR will affect:
Loading of Archived MURs

## Related PRs
List related PRs against other branches:

https://github.com/fecgov/fec-cms/issues/2185

### CODE REVIEW

**Comment by lbeaufort** [comment]
@rjayasekera I thought this needs more work to fully address #3234?

**Comment by rjayasekera** [comment]
@vrajmohan As you suggested I have created 3 separate commits for the 3 type of changes.  @lbeaufort FYI - this issue is far from over. Next step would be to have those multiple PDF files store in the Elastic Search Index.  At this time all the PDF files associated with an archive MUR goes into S3 bucket but they are not indexed.

File: webservices/legal_docs/load_legal_docs.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def get_mur_names(mur_names={}):
                 mur_names[row[1]] = row[2]
     return mur_names
 
-def process_mur(mur):
+def process_murs(raw_mur_tr_element_list):
```
Comment (vrajmohan) [inline webservices/legal_docs/load_legal_docs.py]: This commit does not work. When I run the code, I get:
```
INFO:manager:Loading archived MURs
Traceback (most recent call last):
  File "manage.py", line 381, in <module>
    manager.run()
  File "/Users/radhakrishnanvrajmohan/git/openFEC/.env/lib/python3.6/site-packages/flask_script/__init__.py", line 412, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/Users/radhakrishnanvrajmohan/git/openFEC/.env/lib/python3.6/site-packages/flask_script/__init__.py", line 383, in handle
    res = handle(*args, **config)
  File "/Users/radhakrishnanvrajmohan/git/openFEC/.env/lib/python3.6/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/Users/radhakrishnanvrajmohan/git/openFEC/webservices/legal_docs/load_legal_docs.py", line 446, in load_archived_murs
    process_murs(raw_mur_tr_element_list)
  File "/Users/radhakrishnanvrajmohan/git/openFEC/webservices/legal_docs/load_legal_docs.py", line 385, in process_murs
    = re.findall("<td[^>]*>(.*?)</td>", mur[2], re.S)
NameError: name 'mur' is not defined
```

**Comment by vrajmohan** [comment]
Ideally, commit messages should be shorter - https://chris.beams.io/posts/git-commit/

**Comment by rjayasekera** [comment]
Ok, I'll remember to make the commit messages shorter next time. Thanks.

File: webservices/legal_docs/load_legal_docs.py (hunk 1 above)
Comment (rjayasekera) [reply webservices/legal_docs/load_legal_docs.py]: @vrajmohan I am not sure why and how this works on mine
./manage.py load_archived_murs -s 1252
INFO:manager:Loading archived MURs
INFO:manager:Loading archived MUR 1252
INFO:manager:Loading file 1252_A.pdf
INFO:manager:Loading file 1252_B.pdf
INFO:manager:Loading file 1252_C.pdf
INFO:manager:Loading file 1252_D.pdf
INFO:manager:Loading file 1252_E.pdf
INFO:manager:1 archived MURs loaded

**Comment by vrajmohan** [comment]
It may be a good opportunity to practice `git rebase`. Please try to make the existing commit messages better.

File: webservices/legal_docs/load_legal_docs.py (hunk 1 above)
Comment (vrajmohan) [reply webservices/legal_docs/load_legal_docs.py]: Try running it without specifying a specific archived MUR to load.

**Comment by lbeaufort** [comment]
@vrajmohan @rjayasekera this PR is ready for review. If you'd like me to do a code review, I'd be happy to.

File: tests/test_load_legal_docs.py [hunk 2]
```diff
@@ -165,16 +165,16 @@ def filter(self, Prefix):
         return [o for o in self.objects if o.key.startswith(Prefix)]
 
 class BucketMock:
-    def __init__(self, key):
+    def __init__(self, keys):
```
Comment (vrajmohan) [inline tests/test_load_legal_docs.py]: Good work!

File: webservices/legal_docs/load_legal_docs.py [hunk 3]
```diff
@@ -377,55 +377,80 @@ def get_mur_names(mur_names={}):
...
+            'open_date': open_date,
+            'close_date': close_date,
+            'complainants': complainants,
+            'respondents': respondents,
+        }
+        mur['subject'] = get_subject_tree(subject_td)
+        mur['citations'] = get_citations(re.findall("(.*?)<br>", citations_td))
+        mur['documents'] = get_documents(mur_no_td, bucket)
+
+        # Match the original format for MURs with 1 PDF
```
Comment (vrajmohan) [inline webservices/legal_docs/load_legal_docs.py]: As discussed, we may not need this.

File: webservices/legal_docs/load_legal_docs.py [hunk 4]
```diff
@@ -377,55 +377,80 @@ def get_mur_names(mur_names={}):
...
+    documents = []
+    for index, file_name in enumerate(re.findall(r"/disclosure_data/mur/([0-9_A-Z]+)\.pdf", td_text)):
+        logger.info("Loading file %s.pdf", file_name)
+        pdf_key = 'legal/murs/%s.pdf' % file_name
+        pdf_text, pdf_size, pdf_pages = process_mur_pdf(file_name, pdf_key, bucket)
+        pdf_url = '/files/' + pdf_key
+        document = {
+            "document_id": index + 1,
+            "size": pdf_size,
+            "pages": pdf_pages,
```
Comment (vrajmohan) [inline webservices/legal_docs/load_legal_docs.py]: We can get rid of this field as we don't use it (and don't have it for current MURs).

Comment (lbeaufort) [reply webservices/legal_docs/load_legal_docs.py]: Sounds good. I also realized the size should be labeled `length` for consistency.

File: webservices/legal_docs/load_legal_docs.py [hunk 5]
```diff
@@ -377,55 +377,80 @@ def get_mur_names(mur_names={}):
...
+            if match.group(1) == 'C':
+                complainants.append(name)
+            if match.group(1) == 'R':
+                respondents.append(name)
+
+        mur_name = mur_names.get(mur_no, '')
+        mur = {
+            'doc_id': 'mur_%s' % mur_no,
+            'no': mur_no,
+            'name': mur_name,
```
[the remark this answers is not in the document]
Comment (lbeaufort) [reply webservices/legal_docs/load_legal_docs.py]: Noting here that "old-style" archived MURs have the PDF url in the `url` field, while current MURs have the canonical page link there. I updated this to be the canonical page URL and confirmed that the front end uses `mur.url` for the PDF link for "old" style archived MURs and `mur.document.url` for the "new" style archived MURs.

**Comment by pkfec** [comment]
@rjayasekera can you review this PR before @fec-jli cuts the release branches.

**Pull request merged**